web/src/app/components/cdf/cdf.component.ts

```diff
@@ -0,0 +1,222 @@
+      .domain([0, maxDuration + 1])
+      .rangeRound([0, width])
+      .nice();
+
+    const yScale = d3
+      .scaleLinear()
+      .domain([0, 1])
+      .rangeRound([height, 0]);
+
+    this.graphData = Array.from(this.data);
+    this.graphData.push({
+      duration: xScale.ticks()[xScale.ticks().length - 1],
+      probability: 1
+    });
+
+    this.svg = (d3
+      .select(element)
+      .append('svg') as d3SVG)
+      .attr('width', elementWidth)
+      .attr('height', elementHeight);
+
+    const cdfChart = this.svg
+      .append('g')
+      .attr('id', 'chart')
+      .attr('transform', `translate(${margin.left}, ${margin.top})`);
+
+    const yAxisLeft = cdfChart
+      .append('g')
+      .attr('class', 'y-axis-left')
+      .call(d3
+        .axisLeft(yScale)
+        .ticks(10)
+        .tickFormat(d3.format(',.1f'))
+      );
+
+    // Removes axis's vertical line and keeps the tick marks.
+    yAxisLeft.select('.domain').remove();
+
+    cdfChart.append('text')
+      .attr('class', 'y-axis-left-label')
+      .attr('transform', 'rotate(-90)')
+      .attr('y', -margin.left)
+      .attr('x', -height / 2)
+      .attr('dy', '1em')
+      .style('text-anchor', 'middle')
+      .style('font-size', '12px')
+      .text('Probability');
+
+    const yAxisRight = cdfChart
+      .append('g')
+      .attr('class', 'y-axis-right')
+      .attr('transform', `translate(${width}, 0)`)
+      .call(d3
+        .axisRight(yScale)
+        .ticks(10)
+        .tickFormat(d3.format(',.1f'))
+      );
+
+    // Removes axis's vertical line and keeps the tick marks.
+    yAxisRight.select('.domain').remove();
+
+    const xAxis = this.svg
+      .append('g')
+      .attr('class', 'x-axis')
+      .attr('transform', `translate(${margin.left}, ${height + margin.top})`)
+      .call(d3.axisBottom(xScale));
+
+    this.svg.append('text')
+      .attr('class', 'x-axis-label')
+      .attr('transform',
+            `translate(${(elementWidth) / 2}, ${elementHeight - margin.bottom / 2})`)
+      .style('text-anchor', 'middle')
+      .style('font-size', '12px')
+      .text('Duration (minutes)');
+
+    this.svg.append('text')
+      .attr('class', 'graph-title')
+      .attr('x', elementWidth / 2)
+      .attr('y', margin.top / 2)
+      .attr('text-anchor', 'middle')
+      .style('font-size', '16px')
+      .text('CDF of completed push durations');
+      
+    cdfChart
+      .datum(this.graphData)
+      .append('path')
+      .attr('class', 'cdf-curve')
+      .attr('fill', CdfComponent.COMPLETED_BLUE)
+      .attr('d', d3.area<Item>()
+        .x(d => xScale(d.duration))
+        .y1(d => yScale(d.probability))
+        .y0(yScale(0))
+        .curve(d3.curveStepAfter)
+      );
+  }
+
+  /*
+   * After the component's view has been fully initialized, the chart can be
+   * created. Since the pushInfos are static, we do not need to create the chart
+   * at every change.
+   */
+  ngAfterViewInit(): void { this.createChart(); }
```

Comment:
What I was suggesting is moving the content of `createChart` inside the `ngAfterViewInit`.